rllib/evaluation/collectors/simple_list_collector.py

```diff
@@ -217,10 +334,21 @@ def build(self, view_requirements: ViewRequirementsDict) -> SampleBatch:
             # Create the batch of data from the different buffers.
             data_col = view_req.data_col or view_col
 
-            # Some columns don't exist yet (get created during postprocessing).
-            # -> skip.
             if data_col not in self.buffers:
-                continue
+                # special treatment for state_out_<i>
+                # add them to the buffer in case they don't exist yet
+                if data_col.startswith("state_out_"):
```

Comment:
if this can be shared with build_for_inference(), then we don't need to have this block of duplicated code.

Reply:
done.